antelope_contracts/contracts/erc20/src/erc20.cpp

```diff
@@ -334,4 +334,40 @@ void erc20::removeegress(const std::vector<name>& accounts) {
             egresslist_table.erase(it);
 }
 
+void erc20::setfee(eosio::name token_contract, eosio::symbol token_symbol, const eosio::asset &egress_fee) {
+    require_auth(get_self());
+
+    eosio::check(egress_fee.symbol == native_token_symbol, "egress_fee should have native token symbol");
+
+    uint128_t v = token_contract.value;
+    v <<= 64;
+    v |= token_symbol.code().raw();
+
+    token_table_t token_table(_self, _self.value);
+    auto index_symbol = token_table.get_index<"by.symbol"_n>();
+    auto iter = index_symbol.find(v);
+    check(iter != index_symbol.end(), "token not registered");
+
```

Comment:
Also, check that the symbol (not just the symbol code) matches as well. The symbol code is just the token symbol without the precision, e.g. “USDT”. But the full symbol includes the precision, e.g. “4,USDT”.


You can simply check that token_symbol is equal to the symbol of ingress_fee within the found table entry.

Reply:
Alternatively, the action could just take the symbol code since that is all we care about for finding the table.

Reply:
changed